bitbucket-slack-server-integration-plugin/src/main/java/com/atlassian/bitbucket/plugins/slack/listener/SlackChannelEventListener.java

```diff
@@ -62,10 +67,14 @@ public void onConversationsLoadedEvent(final SlackConversationsLoadedEvent event
         // unmute channel if they are returned in the list of active channels;
         // just in case we missed 'channel_unarchive' event
         asyncExecutor.run(() -> {
-            final List<String> mutedChannelIds = settingService.getMutedChannelIds();
+            final List<ConversationKey> mutedChannelIds = settingService.getMutedChannelIds();
             if (!mutedChannelIds.isEmpty()) {
-                Set<String> activeConversationIds = event.getConversations().stream()
-                        .map(Conversation::getId)
+                Set<ConversationKey> activeConversationIds = event.getConversations().stream()
+                        .flatMap(c -> {
+                            List<ConversationKey> list = new ArrayList<>();
+                            c.getSharedTeamIds().forEach(t -> list.add(new ConversationKey(t, c.getId())));
```

Comment:
Using `getSharedTeamIds` will not work as expected.
The way I think you can get the right teamId here is adding it to the SlackConversationsLoadedEvent object.
This event is created in the DefaultSlackClient class. Then, when instantiating the object there, pass `slackLink.getTeamId()`.
Finally, in this listener, you can do:
```
.map(c -> new ConversationKey(event.getTeamId(), c.getId())))
```